hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferContainer.java

```diff
@@ -71,7 +74,7 @@
      * <li>{@link InMemoryFormat#BINARY} or {@link InMemoryFormat#NATIVE} - the type is {@link Data}</li>
      * </ul>
      */
-    private Ringbuffer ringbuffer;
+    private Ringbuffer<T> ringbuffer;
```

Comment:
The type of the items `Ringbuffer` is not necessarily the same as the type of the container. The javadoc should clarify that so please check it out.

Reply:
I've split up the types to `<T>` for the `RingbufferContainer` and `<E>` for the `Ringbuffer` implementation, where `<T>` is the user type and `<E>` is the internal storage type.